### PR TITLE
fix(tab-title): remove additional error border and prevent content jumping when has siblings FE-4943

### LINKED ISSUE #4833: [ Tabs ] visual issues with the component
### Current behaviour

1. The right border gains the `red` color onHover (with-alternate-styling story) 
2. The focus lefts the small vertical right border when switch to another Tab
3. The inner content of the Tab just goes down (validation story)
4. The error border goes outside of the `Tab` component

### Expected behaviour

1. The right border should not gain the `red` color onHover (with-alternate-styling story) 
2. The focus doesn't left the small vertical right border when switch to another Tab
3. The inner content of the Tab doesn't go down (validation story)
4. The error border doesn't go outside of the `Tab` component

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/story/tabs--default-story

### JIRA Ticket (Sage Only)

FE-4943, FE-4944, FE-4945

### Suggested Solution

_No response_

### Carbon Version

104.15.0

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari

### What Operating System are you seeing the problem on?

MacOS

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### LINKED ISSUE #4875: Color of icons in Tabs
### Current behaviour

The color of an icon in the customLayout of a Tab is not correct

### Expected behaviour

If i use the color prop on a icon, the icon should be displayed with that color

### CodeSandbox or Storybook URL

https://codesandbox.io/s/jolly-fire-3wcfj6

### JIRA Ticket (Sage Only)

_No response_

### Suggested Solution

_No response_

### Carbon Version

104.21.1

### What browsers are you seeing the problem on?

Firefox, Chrome

### What Operating System are you seeing the problem on?

Windows

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by edleeks87**
fix #4833
fix #4875

### Proposed behaviour

Updates `TabTitle` styling to remove additional right border when error state detected
image

Prevents content jumping when `siblings` prop used and `isTabSelected`.
image

Updates styling to remove min-width when `customLayout` is used and updates demo to use `Box` and `flex` layout.
image

### Current behaviour

Extra border-right is added when error state and alternateStyling is used (alternate-styling story).
image

The content jumps when a tab with sibling elements is selected.
image

The `customLayout` story uses grid layout and sets the container `min-width: 100px` meaning it overflows the `TabTitle` when error state is active.
image

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] Unit tests added or updated if required
- [x] Storybook added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

Check Tab stories on storybook

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

**Review by grabkowski — DISMISSED** [review]

File: src/components/tabs/__internal__/tab-title/tab-title.style.js [hunk 1]
```diff
@@ -14,9 +14,7 @@ const StyledTitleContent = styled.span`
     position,
     noLeftBorder,
     noRightBorder,
-    isTabSelected,
     hasHref,
-    error,
```
Comment (mkrds) [inline src/components/tabs/__internal__/tab-title/tab-title.style.js:19]: I think you should also remove these props from <StyledTitleContent ...> in src/components/tabs/__internal__/tab-title/tab-title.component.js

File: src/components/tabs/__internal__/tab-title/tab-title.style.js [hunk 2]
```diff
@@ -212,28 +201,12 @@ const StyledTitleContent = styled.span`
...
-    hasSiblings,
-    hasCustomLayout,
-    error,
-    warning,
-    info,
-    size,
-    isTabSelected,
-    position,
-  }) =>
+  ${({ hasSiblings, hasCustomLayout, position }) =>
```
Comment (mkrds) [inline src/components/tabs/__internal__/tab-title/tab-title.style.js:204]: Same as above, I think you should also remove these props from <StyledTitleContent ...> in src/components/tabs/__internal__/tab-title/tab-title.component.js

File: src/components/tabs/__internal__/tab-title/tab-title.style.js (hunk 1 above)
Comment (edleeks87) [reply src/components/tabs/__internal__/tab-title/tab-title.style.js:19]: They are actually used still but destructured later in the file, I'll refactor it so it's done at the top level

**Review by grabkowski — DISMISSED** [review]

**Review by mkrds — DISMISSED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**